### ISSUE TITLE
Cannot link minimal_surface

### ISSUE THREAD

**Issue by GH-HOME** [zhenzhangye/general_ups#1]
hi, I clone the minimal_surface and normal integration in the third_part folder and I am able to run the demo of these two projects. However the naturalUPS.m still cannot link the two projects. I try to modify line 18-20 in naturalUPS.m, but still cannot run the code.

Could you please help me figure out how to link minimal_surface in naturalUPS?

Thanks.

**Comment by BjoernHaefner** [comment]
Hi @GH-HOME, it's a good sign that you can run the [minimalSurface](https://github.com/zhenzhangye/minimal_surface) code. So this works. Then there seems to be a linking problem. What's the error message you get?

Did you add the minimalSurface folder to the search path? We already do this [here](https://github.com/zhenzhangye/general_ups/blob/f7389f798fe5cab586574ce6b885a049bfd146b9/main.m#L10), but maybe your folder structure is different?

**Comment by GH-HOME** [comment]
Hi, @BjoernHaefner, thanks for your reply. 
The folder structure is 
- third_party
   - minimal_surface
   - normal_integration

The error message is:

Error using naturalUPS (line 26)
No depth initialization!

I'm a little confused why line 18~20 in naturalUPS.m is commented. But if I uncomment these code, the error is "Undefined function or variable 'minimalSurface'.

Thanks.

**Comment by BjoernHaefner** [comment]
Hi @GH-HOME,

and third_party is inside the `general_ups` folder? If yes, then this should be alright.
Yeah, we had a fresh commit to master that uncomment these lines, i.e. l.18-20 in `main.m` should **not** be commented. It might be best if you update your repo with the latest changes.

Where does the `MinimalSurfaceMEX.mexa64` function (the mexFile) reside after building MinimalSurface? This is the function that should be in the search tree of Matlab. Add the path to the mexFile in the `main.m` file, i.e. if the `MinimalSurfaceMEX.mexa64` file is in `minimalsurface/build/lib`, then in `main.m` write `addpath(path/to/minimalsurface/build/lib);` after `clear all; close all; clc;` and before you call the `naturalUPS/generalUPS` function. (We renamed the function from `naturalUPS` to `generalUPS` in the latest commit). Also make sure that you don't call `clear all` after adding the path to the mex file. If all of this doesn't work, try to run `restoredefaultpath` in the command window before running `main.m`

**NOTE:** You also might have to clone [this repo](https://github.com/zhenzhangye/orthonormal_to_perspective) for normal integration. We updated the readme accordingly.
